src/core/viz/expressions/torque.js

```diff
@@ -203,10 +210,99 @@ export class Torque extends BaseExpression {
         return new Date(tmix);
     }
 
+    /**
+     * Set the time stamp of the simulation
+     * @api
+     * @memberof carto.expressions.Torque
+     * @instance
+     * @name setSimTime
+     * @param {Date|number} simulationTime - A javascript Date object with the new simulation time
+     */
+    setSimTime(simulationTime) {
+        simulationTime = castDate(simulationTime);
+        
+        const tmin = this._input.min.eval();
+        const tmax = this._input.max.eval();
+
+        if (simulationTime.getTime() < tmin) {
```

Comment:
I would assign `simulationTime.getTime()` to a variable to avoid calling this method three times.

Reply:
No, please, no. "Premature optimization is the root of all evil"